zerver/worker/queue_processors.py

```diff
@@ -115,25 +115,77 @@ def on_failure(event: Dict[str, Any]) -> None:
 
 class QueueProcessingWorker(ABC):
     queue_name = None  # type: str
+    CONSUME_ITERATIONS_BEFORE_UPDATE_STATS_NUM = 2
```

Comment:
I assume the idea is this will be 50 in the real world?  We could put it in `settings` and then override in `dev_settings` and `test_settings`, perhaps?

Reply:
Yes, I was thinking something like 50 default. Alternatively to the settings approach, it may make sense to override this on the various queue classes, as different queues may have a different good value for this? A queue that processes a lot of events very quickly perhaps should have this higher than a queue  that processes fewer events that take a lot of time to process?